File: code/11.container-with-most-water.php
<?php
/*
 * @lc app=leetcode id=11 lang=php
 *
 * [11] Container With Most Water
 */

// @lc code=start
class Solution {

    /**
     * @param Integer[] $height
     * @return Integer
     */
    function maxArea($height) {
        $i = 0;
        $j = count($height) - 1;
        $result = 0;
        while ($i < $j) {
            $x = $j - $i;
            $y = min($height[$i], $height[$j]);
            $result = max($result, $x * $y);

            if ($height[$i] <= $height[$j]) {
                $i++;
            } else {
                $j--;
            }
        }

        return $result;
    }
}
// @lc code=end

File: code/11.container-with-most-water.go
package code

/*
 * @lc app=leetcode id=11 lang=golang
 *
 * [11] Container With Most Water
 */

// @lc code=start
func maxArea(height []int) int {
	i, j := 0, len(height) - 1

	result := 0
	for i < j {
		s := min(height[i], height[j]) * (j - i)
		if s > result {
			result = s
		}

		if height[i] < height[j] {
			i++
		} else {
			j--
		}
	}

	return result
}
// @lc code=end

